torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -654,50 +679,86 @@ def pad_image_tensor(
     fill: features.FillTypeJIT = None,
     padding_mode: str = "constant",
 ) -> torch.Tensor:
+    # Be aware that while `padding` has order `[left, top, right, bottom]` has order, `torch_padding` uses
+    # `[left, right, top, bottom]`. This stems from the fact that we align our API with PIL, but need to use `torch_pad`
+    # internally.
+    torch_padding = _parse_pad_padding(padding)
+
+    if padding_mode not in ["constant", "edge", "reflect", "symmetric"]:
+        raise ValueError(
+            f"`padding_mode` should be either `'constant'`, `'edge'`, `'reflect'` or `'symmetric'`, "
+            f"but got `'{padding_mode}'`."
+        )
+
     if fill is None:
         # This is a JIT workaround
-        return _pad_with_scalar_fill(image, padding, fill=None, padding_mode=padding_mode)
-    elif isinstance(fill, (int, float)) or len(fill) == 1:
-        fill_number = fill[0] if isinstance(fill, list) else fill
-        return _pad_with_scalar_fill(image, padding, fill=fill_number, padding_mode=padding_mode)
+        return _pad_with_scalar_fill(image, torch_padding, fill=0, padding_mode=padding_mode)
```

Comment:
Why `fill=0` instead of `fill=None` as originally ? Do we still need this workaround ?

Reply:
Previously, this was handled by `_FT.pad`:

https://github.com/pytorch/vision/blob/e64784cdea465d833d9d0f66dc73d7abe217933d/torchvision/transforms/functional_tensor.py#L386-L387

Since we are no longer calling it, we need to handle it our own. For some reason, the fill overwrite above did not work for me while developing and so I went this way. Rechecking today, it seems to work and it was probably caused by something else. I've fixed this in 1622cd6c0ca320ac7acb1b909924cc86db4af936.

More general though, why are we allowing `None` in the first place and even use it as default if we all we do with it is to map it to `0`? I faintly remember there were discussions about it, but I think I was out of the loop on them. Is there a public discussion that you can point me to or did this happen offline? Intuitively, I would remove the `None` as valid value and just use `0` as default.

Reply:
> More general though, why are we allowing None in the first place and even use it as default if we all we do with it is to map it to 0? 

Origins for fill=None starts from your PR https://github.com/pytorch/vision/pull/1760 on rotate op.
See this https://github.com/pytorch/vision/issues/6623 for later discussions